src/flows/recommendation_api/ff_new_tab_aggregated_engagement.py

```diff
@@ -0,0 +1,108 @@
+            COUNT(*) AS TRAILING_1_DAY_IMPRESSIONS,
+            SUM(CASE WHEN click IS NOT NULL THEN 1 ELSE 0 END) AS TRAILING_1_DAY_OPENS,
+            -- For now, we only need 1 day trailing data, so leave the other ones at 0.  
+            0 AS TRAILING_7_DAY_IMPRESSIONS,
+            0 AS TRAILING_7_DAY_OPENS,
+            0 AS TRAILING_14_DAY_IMPRESSIONS,
+            0 AS TRAILING_14_DAY_OPENS,
+            0 AS TRAILING_21_DAY_IMPRESSIONS,
+            0 AS TRAILING_21_DAY_OPENS,
+            0 AS TRAILING_28_DAY_IMPRESSIONS,
+            0 AS TRAILING_28_DAY_OPENS
+        FROM impressions_data
+        CROSS JOIN UNNEST(impressions_data.tiles) AS flattened_tiles
+        GROUP BY TILE_ID
+        ORDER BY TRAILING_1_DAY_IMPRESSIONS DESC
+        LIMIT 16384 -- Limit to Snowflake's max list size. There are a lot of old items that still get some impressions.
+    """
+
+
+EXPORT_CORPUS_ITEM_KEYS_SQL = """
+    SELECT DISTINCT 
+        TILE_ID,
+        concat_ws(
+            -- corpus-engagement-v1 is keyed on the following three fields, separated by slashes.
+            '/', recommendation_surface_id, corpus_slate_configuration_id, corpus_item_id
+        ) as KEY,
+        RECOMMENDATION_SURFACE_ID,
+        CORPUS_SLATE_CONFIGURATION_ID,
+        CORPUS_ITEM_ID
+    FROM ANALYTICS.DBT_STAGING.STG_CORPUS_SLATE_RECOMMENDATIONS
+    WHERE TILE_ID in (%(tile_ids)s)
+"""
+
+
+@task()
+def df_column_to_list(df: pd.DataFrame, column_name: str):
+    return df[column_name].tolist()
+
+
+@task()
+def pd_merge(left: pd.DataFrame, right: pd.DataFrame) -> pd.DataFrame:
+    return pd.merge(left, right)
+
+
+@task()
+def df_drop(df: pd.DataFrame, columns: Union[str, List[str]]) -> pd.DataFrame:
+    return df.drop(columns=columns)
+
+
+@task()
+def df_is_empty(df: pd.DataFrame) -> bool:
+    return df.empty
+
+
+with Flow(FLOW_NAME, schedule=get_interval_schedule(minutes=15)) as flow:
+    # Get telemetry from BigQuery
+    df_telemetry = BigQueryTask()(
+        query=EXPORT_FIREFOX_TELEMETRY_SQL,
+        to_dataframe=True,
+    )
+
+    # For the tileIds from BigQuery, get the metadata required by corpus-engagement-v1 that's stored in Snowflake.
+    df_corpus_item_keys = PocketSnowflakeQuery()(
+        query=EXPORT_CORPUS_ITEM_KEYS_SQL,
+        data={'tile_ids': df_column_to_list(df_telemetry, column_name='TILE_ID')},
+    )
+
+    # If none of the tileIds exist in Snowflake, then there's nothing to do. This should only happen while the new API
+    # is under development, and there are days without any NewTab impressions on content served from Recommendation API.
+    with case(df_is_empty(df_corpus_item_keys), False):
+        # Combine the BigQuery and Snowflake results on TILE_ID.
+        df_keyed_telemetry = pd_merge(df_telemetry, df_corpus_item_keys)
+        # Drop TILE_ID now we no longer need it, to match the dataframe columns with the feature group.
+        df_keyed_telemetry = df_drop(df_keyed_telemetry, columns=['TILE_ID'])
```

Comment:
I kept Pandas functions as simple tasks for readability, instead of combining pd_merge and df_drop.